mission_control/src/mission_control.cpp

```diff
@@ -271,167 +195,112 @@ void MissionControlNode::loadMissionCallback(const mission_control::LoadMission:
-
-  for (it = m_mission_map.begin(); it != m_mission_map.end(); it++)
+  if (m_mission_map.size() > 0)
   {
-    if ((it->second->GetState() != Mission::MissionState::ABORTING) &&
-        (it->second->GetState() != Mission::MissionState::EXECUTING))
+    if (m_current_mission_id != 0)
     {
-      ROS_INFO("Deleting mission id[%d] with description[%s]", it->first,
-               (it->second->getMissionDescription()).c_str());
-      delete it->second;
-      m_mission_map.erase(it);
+      executeMissionTimer.stop();
+      m_mission_map[m_current_mission_id]->stop();
+      m_current_mission_id = 0;
+      m_mission_id_counter = 0;
     }
-  }
-}
-
-void MissionControlNode::correctedDataCallback(const pose_estimator::CorrectedData& data)
-{
-  // mission_map.count checks to see if a key is in the map
-  if ((m_current_mission_id > 0) && (m_mission_map.size() > 0) &&
-      (m_mission_map.count(m_current_mission_id) > 0))
-  {
-    m_mission_map[m_current_mission_id]->ProcessCorrectedPoseData(data);
+    ROS_DEBUG_STREAM("removeMissionsCallback - Removing missions");
+    m_mission_map.clear();
   }
 }
 
 void MissionControlNode::reportFaultCallback(const health_monitor::ReportFault::ConstPtr& msg)
 {
   if (msg->fault_id != 0)
   {
-    ROS_WARN_STREAM("Fault Detected [" << msg->fault_id << "] aborting mission");
+    ROS_ERROR_STREAM("Fault Detected by health monitor[" << msg->fault_id << "] aborting mission");
     abortMission(m_current_mission_id);
   }
 }
 
-bool MissionControlNode::endsWith(const std::string& str, const char* suffix)
-{
-  unsigned suffixLen = std::string::traits_type::length(suffix);
-  return str.size() >= suffixLen &&
-         0 == str.compare(str.size() - suffixLen, suffixLen, suffix, suffixLen);
-}
-
-bool MissionControlNode::FoundMissionFile()
+void MissionControlNode::processAbort()
 {
-  ROS_INFO("Mission path is [%s]", mission_path.c_str());
-
-  bool retval = false;
-  DIR* d;
-  struct dirent* dir;
-  d = opendir(mission_path.c_str());
-
-  if (d)  // if the directory is valid
-  {
-    if (endsWith(mission_path, ".xml")) return true;
-
-    while (retval == false)
-    {
-      while ((dir = readdir(d)) != NULL)  // read file in directory
-      {
-        // Condition to check regular file.
-        if (dir->d_type == DT_REG)
-        {
-          mission_path.append("/");
-          mission_path.append(dir->d_name);
-
-          ROS_INFO("Found mission file. full path is: %s\n", mission_path.c_str());
-          retval = true;
-          break;
-        }
-      }
-
-      if (retval == false)
-      {
-        ROS_INFO("sleeping");
-        usleep(250000);
-      }
-    }
-  }
-  else
-  {
-    ROS_ERROR("The directory path to the mission file is invalid: %s", mission_path.c_str());
-  }
-
-  closedir(d);
-  return retval;
+  jaus_ros_bridge::ActivateManualControl msg;
+  msg.activate_manual_control = true;
+  pub_activate_manual_control.publish(msg);
```

Comment:
@patapufate ???

Why is the `mission_control` using a user interface to abort missions?

Reply:
After meeting. At this moment we are implementing change the autopilot to manual mode. this behavior set the trhuster to 0 RPM and set fins to surface